### PR TITLE
[rush] install-run doesn't respect RUSH_TEMP_FOLDER environment variable

### LINKED ISSUE #1431: [rush] install-run.js doesn't respect RUSH_TEMP_FOLDER
## Please prefix the issue title with the project name i.e. [rush], [api-extractor] etc. ##

**Is this a feature or a bug?**

- [x] Feature
- [ ] Bug

**Please describe the actual behavior.**
This is sort-of a feature, but for me it acts more like a bug.

running `install-run-rush.js` with `RUSH_TEMP_FOLDER` set installs rush into `common/temp` and then runs rush, which then respects `RUSH_TEMP_FOLDER`. The rush executable works, and rush's installs work, but the directories are different.

**If the issue is a bug, how can we reproduce it? Please provide detailed steps and include a GitHub branch if applicable. Your issue will get resolved faster if you can make it easy to investigate.**

Setting `RUSH_TEMP_FOLDER` to any other directory, and then running `install-run.js`.
Personally, I'm having trouble because I'm using `install-run-rush.js` to run rush, which installs rush into the wrong temp folder, and then starts respecting the environment variable.

In `common/scripts/install-run.js`:
`const rushTempFolder = ensureAndJoinPath(rushCommonFolder, 'temp');`
and
`const packageInstallFolder = ensureAndJoinPath(rushCommonFolder, 'temp', ...`
are to blame here.

**What is the expected behavior?**

I'd expect `install-run.js` to respect `RUSH_TEMP_FOLDER`.

**If this is a bug, please provide the tool version, Node.js version, and OS.**

* **Tool:** rush
* **Tool Version:** 5.10.3
* **Node Version:** (shouldn't be relevant, things are working)
  * **Is this a LTS version?** 
  * **Have you tested on a LTS version?** 
* **OS:** Ubuntu 16.04

### PR DESCRIPTION
**Pull request by nayanshah**
Fixes #1431

### CODE REVIEW

File: common/changes/@microsoft/rush/intall-run-temp_2019-07-31-21-34.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
+  "changes": [
+    {
+      "comment": "install-run doesn't respect RUSH_TEMP_FOLDER environment variable",
```
Comment (iclanton) [inline common/changes/@microsoft/rush/intall-run-temp_2019-07-31-21-34.json]:
```suggestion
      "comment": "Add support for the RUSH_TEMP_FOLDER environment variable in the install-run-rush script.",
```

File: apps/rush-lib/src/scripts/install-run.ts [hunk 2]
```diff
@@ -19,6 +19,7 @@ import * as path from 'path';
 import { IPackageJson } from '@microsoft/node-core-library';
 
 export const RUSH_JSON_FILENAME: string = 'rush.json';
+const RUSH_TEMP_FOLDER: string = 'RUSH_TEMP_FOLDER';
```
Comment (iclanton) [inline apps/rush-lib/src/scripts/install-run.ts]:
```suggestion
const RUSH_TEMP_FOLDER_ENV_VARIABLE_NAME: string = 'RUSH_TEMP_FOLDER';
```

File: apps/rush-lib/src/scripts/install-run.ts [hunk 3]
```diff
@@ -371,6 +371,10 @@ function writeFlagFile(packageInstallFolder: string): void {
   }
 }
 
+function getRushTempFolder(rushCommonFolder: string): string {
+  return process.env[RUSH_TEMP_FOLDER] || ensureAndJoinPath(rushCommonFolder, 'temp');
```
Comment (iclanton) [inline apps/rush-lib/src/scripts/install-run.ts]: We should also ensure the folder we get from the env variable exists. See the logic here: https://github.com/microsoft/web-build-tools/blob/master/apps/rush-lib/src/api/EnvironmentConfiguration.ts#L112

Comment (nayanshah) [reply apps/rush-lib/src/scripts/install-run.ts]: Added a check to make sure the folder exists.

**Comment by nayanshah** [comment]
@iclanton : I've addressed feedback for this one.

File: apps/rush-lib/src/scripts/install-run.ts [hunk 4]
```diff
@@ -371,6 +371,19 @@ function writeFlagFile(packageInstallFolder: string): void {
   }
 }
 
+function getRushTempFolder(rushCommonFolder: string): string {
+  const rushTempFolder: string | undefined = process.env[RUSH_TEMP_FOLDER_ENV_VARIABLE_NAME];
+  if (rushTempFolder !== undefined) {
+    if (!fs.existsSync(rushTempFolder)) {
```
Comment (iclanton) [inline apps/rush-lib/src/scripts/install-run.ts]: This will only work if the only non-existant folder is the leaf folder.

Comment (iclanton) [reply apps/rush-lib/src/scripts/install-run.ts]: Made a quick change to address it.

**Pull request merged**